src/js/common/schemaform/helpers.js

```diff
@@ -420,7 +427,11 @@ export function setHiddenFields(schema, uiSchema, formData) {
   }
 
   if (updatedSchema.type === 'array') {
-    const newSchema = setHiddenFields(updatedSchema.items, uiSchema.items, formData);
+    // Our current approach is to sometimes modify the schema based on form data
+    // This breaks down with arrays, because all items share the same schema
+    // So, this is technically not correct, but the last row is probably what's visible
+    // for a user, so it should be the least bad for now
+    const newSchema = setHiddenFields(updatedSchema.items, uiSchema.items, formData, path.concat(formData.length - 1));
```

Comment:
Correct. I don't think we have hidden fields inside of arrays right now, so I don't think it was causing any problems. But we do need to fix it at some point.